webapp/src/components/link_tooltip/link_tooltip.jsx

```diff
@@ -0,0 +1,165 @@
+import React, {useEffect, useState} from 'react';
+import PropTypes from 'prop-types';
+import './tooltip.css';
+import Octicon, {GitMerge, GitPullRequest, IssueClosed, IssueOpened} from '@primer/octicons-react';
+import ReactMarkdown from 'react-markdown';
+
+import Client from 'client';
+import {getLabelFontColor} from '../../utils/styles';
+
+export const LinkTooltip = ({href}) => {
+    const [data, setData] = useState(null);
+
+    useEffect(() => {
+        const init = async () => {
```

Comment:
Correct me if I am wrong, but this piece of code will:
1. Execute on component mount
2. Fetch the PR
3. Update the status
4. Render the element with updated status
5. Execute again after the render
6. Fetch the PR one more time
7. Update the status
8. Will not render since the status will be the same as previous.

Maybe a check like `if data != null return` would avoid the second fetch?

Reply:
Hey @larkox, thanks for the feedback. I'm not sure if I follow you... Are you referring to this [check](https://github.com/mattermost/mattermost-plugin-github/pull/258/files#diff-030a588562eadd14c868ae7644412561R80)?

Reply:
That check will only render if there is some data, which makes sure the tooltip does not render before you have retrieved the data from the server, and not render the tooltip if the link is not a github link.

My problem is that the effect you are registering with useEffect, if I am not mistaken, will be executed after every render. I think (correct me if I am wrong) that the body of the effect should only be called once (you only want to fetch the PR once during the lifetime of the tooltip). Having some kind of check to make sure it is only executed once would be nice to have.

Reply:
I see! Great catch. I've just added the line that fixes that :) @larkox